connectors/byoc.py

```diff
@@ -596,3 +598,82 @@ async def sync(self, elastic_server, idling, sync_now=False):
         finally:
             self._syncing = False
             self._start_time = None
+
+
+STUCK_JOBS_THRESHOLD = 60  # 60 seconds
+
+
+class SyncJobIndex(ESIndex):
+    """
+    Represents Elasticsearch index for sync jobs
+
+    Args:
+        index_name (str): Name of an Elasticsearch index
+        elastic_config (dict): Elasticsearch configuration and credentials
+    """
+
+    def __init__(self, index_name, elastic_config):
+        super().__init__(index_name=index_name, elastic_config=elastic_config)
+
+    def _create_object(self, doc_source):
+        """
+        Args:
+            doc_source (dict): A raw Elasticsearch document
+        Returns:
+            SyncJob
+        """
+        return SyncJob(
+            self,
+            connector_id=doc_source["_source"]["connector"]["id"],
+            doc_source=doc_source["_source"],
+        )
+
+    def pending_job_query(self, connectors_ids):
+        """
+        Args:
+            connectors_ids (list of int): A list of connectors IDs
+        Returns:
+            dict
+        """
+        status_term = {"status": [e2str(JobStatus.PENDING)]}
+
+        query = {
+            "bool": {
+                "must": [
+                    {"terms": status_term},
+                    {"terms": {"connector.id": connectors_ids}},
+                ]
+            }
+        }
+
+        return query
+
+    def orphaned_jobs_query(self, connectors_ids):
+        """
+        Args:
+            connectors_ids (list of int): A list of connectors IDs
+        Returns:
+            dict
+        """
+        query = {"bool": {"must_not": {"terms": {"connector.id": connectors_ids}}}}
+
+        return query
+
+    def stuck_jobs_query(self, connectors_ids):
+        """
+        Args:
+            connectors_ids (list of int): A list of connectors IDs
+        Returns:
+            dict
+        """
+        query = {
+            "bool": {
+                "filter": [
+                    {"terms": {"connector.id": connectors_ids}},
+                    {"terms": {"status": [JobStatus.IN_PROGRESS, JobStatus.CANCELING]}},
+                    {"range": {"last_seen": {"lte": "now-#{STUCK_JOBS_THRESHOLD}s"}}},
```

Comment:
you forgot to make this an f-string

Reply:
Omg I saw this too but my brain did not process that that's Ruby code :D

Reply:
lol, my ruby-washed brain :)